pyCGM_Single/pycgmStatic.py

```diff
@@ -1850,123 +1854,122 @@ def rotaxis_footflat(frame,ankle_JC,vsk=None):
+    ...                        [ -0.96,  -0.26,   0.13, 219.47],
+    ...                        [  0.1 ,   0.13,   0.99,  80.63],
+    ...                        [  0.  ,   0.  ,   0.  ,   1.  ]]])
+    >>> [np.around(arr, 2) for arr in calc_axis_nonflatfoot(rtoe, ltoe, rhee, lhee, ankle_axis)] #doctest: +NORMALIZE_WHITESPACE
+    [array([[ -0.1 ,   0.07,   0.99, 442.82],
+            [ -0.94,   0.32,  -0.12, 381.62],
+            [ -0.33,  -0.95,   0.03,  42.66],
+            [  0.  ,   0.  ,   0.  ,   1.  ]]), 
+     array([[  0.12,   0.06,   0.99,  39.44],
+            [ -0.94,  -0.3 ,   0.13, 382.45],
+            [  0.31,  -0.95,   0.03,  41.79],
+            [  0.  ,   0.  ,   0.  ,   1.  ]])]
     """
 
     #REQUIRED MARKERS:
     # RTOE
     # LTOE
-    # ankle_JC
-    TOE_R = frame['RTOE']
-    TOE_L = frame['LTOE']
-    HEE_R = frame['RHEE']
-    HEE_L = frame['LHEE']
+    # RHEE
+    # LHEE
+    # ankle_axis
 
-    ankle_JC_R = ankle_JC[0]
-    ankle_JC_L = ankle_JC[1]
-    ankle_flexion_R = ankle_JC[2][0][1]
-    ankle_flexion_L = ankle_JC[2][1][1]
+    rtoe, ltoe, rhee, lhee, ankle_axis = map(np.asarray, [rtoe, ltoe, rhee, lhee, ankle_axis])
 
-    # Toe axis's origin is marker position of TOE
-    R = TOE_R
-    L = TOE_L
+    ankle_jc_right = ankle_axis[0][:3, 3]
+    ankle_jc_left = ankle_axis[1][:3, 3]
+    ankle_flexion_right = ankle_axis[0][1, :3] + ankle_jc_right
+    ankle_flexion_left = ankle_axis[1][1, :3] + ankle_jc_left
 
-    ankle_JC_R = [ankle_JC_R[0],ankle_JC_R[1],ankle_JC_R[2]]
-    ankle_JC_L = [ankle_JC_L[0],ankle_JC_L[1],ankle_JC_L[2]]
+    # Toe axis's origin is marker position of TOE
+    right_origin = rtoe
+    left_origin = ltoe
 
     # in case of non foot flat we just use the HEE marker
-    R_axis_z = [HEE_R[0]-TOE_R[0],HEE_R[1]-TOE_R[1],HEE_R[2]-TOE_R[2]]
-    R_axis_z = R_axis_z/norm3d(R_axis_z)
+    right_axis_z = [rhee[0]-rtoe[0],rhee[1]-rtoe[1],rhee[2]-rtoe[2]]
+    right_axis_z = right_axis_z/norm3d(right_axis_z)
 
-    y_flex_R = [ankle_flexion_R[0]-ankle_JC_R[0],ankle_flexion_R[1]-ankle_JC_R[1],ankle_flexion_R[2]-ankle_JC_R[2]]
+    y_flex_R = ankle_flexion_right - ankle_jc_right
     y_flex_R = y_flex_R/norm3d(y_flex_R)
 
-    R_axis_x = cross(y_flex_R,R_axis_z)
-    R_axis_x = R_axis_x/norm3d(R_axis_x)
+    right_axis_x = cross(y_flex_R,right_axis_z)
+    right_axis_x = right_axis_x/norm3d(right_axis_x)
 
-    R_axis_y = cross(R_axis_z,R_axis_x)
-    R_axis_y = R_axis_y/norm3d(R_axis_y)
-
-    R_axis_x = [R_axis_x[0]+R[0],R_axis_x[1]+R[1],R_axis_x[2]+R[2]]
-    R_axis_y = [R_axis_y[0]+R[0],R_axis_y[1]+R[1],R_axis_y[2]+R[2]]
-    R_axis_z = [R_axis_z[0]+R[0],R_axis_z[1]+R[1],R_axis_z[2]+R[2]]
+    right_axis_y = cross(right_axis_z,right_axis_x)
+    right_axis_y = right_axis_y/norm3d(right_axis_y)
 
-    R_foot_axis = [R_axis_x,R_axis_y,R_axis_z]
+    right_foot_axis = [right_axis_x,right_axis_y,right_axis_z]
 
     # Left
+    left_axis_z = [lhee[0]-ltoe[0],lhee[1]-ltoe[1],lhee[2]-ltoe[2]]
```

Comment:
`lhee-ltoe` should work, no?

Reply:
I initially changed it to `lhee - ltoe` but for whatever reason it caused a lot of error and most tests failed

Reply:
can you try again, and check what the output is before and after using that? it shouldn't be causing a problem, because it is cast to numpy array before.  otherwise, there is some function later that doesnt like `left_axis_z` being a numpy array

Reply:
Pushed `rhee - rtoe` and `lhee - ltoe` and all is working!